attelo/decoding/control.py

```diff
@@ -60,10 +60,14 @@ def _predict_relate(dpack, models):
     and an list of labels
     """
     if models.relate == 'oracle':
-        idxes = dpack.target
+        unrelated = dpack.label_number(UNRELATED)
         # pylint: disable=no-member
-        probs = np.ones(idxes.shape)
+        # treat unrelated as unknown (to avoid cutting links that may
+        # have been inserted by an non-oracle attachment label)
+        always_related = np.vectorize(lambda x: 0 if x == unrelated else x)
```

Comment:
It does not seem to me that they literally are probabilities here. We haven't used classifiers that do not output probabilities, e.g. sklearn's perceptron, for labelling yet but it makes perfect sense to try that too.